packages/grid/_modules_/grid/GridComponent.tsx

```diff
@@ -129,8 +136,10 @@ export const GridComponent = React.forwardRef<HTMLDivElement, GridComponentProps
             >
               <ApiContext.Provider value={apiRef}>
                 <OptionsContext.Provider value={gridState.options}>
-                  {customComponents.headerComponent || (
-                    <GridToolbar>
+                  {customComponents.headerComponent ? (
```

Comment:
So we can't have a headerComponent and a toolbar at the same time 🤔 
Also we are moving the HeaderRef, not ideal, how about we keep it as a container so around either the toolbar or the header?

Reply:
For the first point - yes you can either have a header or a toolbar, that was decided on the previous PR, the one for the toolbar.

On the second point - sure I can wrap them in a div and add the ref to it. Btw there wasn't a `headerRef` until now. This PR adds it.